cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -0,0 +1,83 @@
+package org.bf2.cos.fleetshard.operator.connector;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.Properties;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+public class ConnectorConfiguration<S> {
+
+    private final static String PROPERTY_DATA_SHAPE = "data_shape";
+    private final static String PROPERTY_PROCESSORS = "processors";
+    private final static String PROPERTY_ERROR_HANDLER = "error_handler";
+    private static final Set<String> RESERVED_PROPERTIES = Set.of(PROPERTY_DATA_SHAPE, PROPERTY_PROCESSORS,
+        PROPERTY_ERROR_HANDLER);
+    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+    private final S connectorSpec;
+    private final ObjectNode dataShapeSpec;
+    private ObjectNode errorHandlerSpec;
+    private ArrayNode processorsSpec = JSON_MAPPER.createArrayNode();
+
+    @SuppressWarnings("unchecked")
+    public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
+        if (null == connectorSpec || connectorSpec.isEmpty()) {
+            throw new RuntimeException("Connector spec can't be empty!");
+        }
+
+        var dataShape = connectorSpec.at("/" + PROPERTY_DATA_SHAPE);
+        if (dataShape.isMissingNode()) {
+            throw new RuntimeException("Missing `data_shape` in connector spec: " + connectorSpec.toPrettyString());
```

Comment:
I assume `data_shape` is mandatory now?

Reply:
I would defer the validation to a future version, at least for camel there are still connectors for which we have do decide what is the right thing to do.

Reply:
According to the Epic Brief `data_shape` should be mandatory. For sinks you it should be mandatory because you have to define the message format of messages coming from Kafka and for sources it should be mandatory to define the format how messages are written into Kafka?

Reply:
yes it should be mandatory but not all the connectors for camel have a proper data shape definition (yet). 
we should also discuss who is setting the defaults i.e. should the UI/fleet-manager set the defaults if they are not provided ? 

So I would track an issue to enable strict checking at a later stage and maybe for the moment, just lo a warning/info.
Would that make sense ?

Reply:
Changed it to only log a warning and added a `TODO` for later.